core/pom.xml

```diff
@@ -106,6 +106,26 @@ THE SOFTWARE.
       <groupId>com.github.jnr</groupId>
       <artifactId>jnr-posix</artifactId>
     </dependency>
+    <dependency>
```

Comment:
@basil what was the reasoning to add these dependencies here now?
The versions should be aligned from the bom change in the same PR, so should not need to be directly added to the dependencies to get the right version?  I doubt we are directly using all of them - so would only expect a dependency on something we directly used in the code rather than anything that was also transitive?

Reply:
until it's fully split it's currently part of the API really.

plugins like token macro depend on ASM. and I think scm-api does as well.

Reply:
I am not following why we add these specifically yet are not adding the guice dependencies that some plugins also need (and blow up spectacurly if they get differing version in a plugin).

in other words there it appears to me we have a lack of consistency on what dependencies we add and what we do.

or to put it another way: why is ok for artifact-manager-s3 to use the bom to get a bundle a compatable version of `guice-inject-assist` but not for `token-macro` to use to get the right version of the `asm` library it requires?

Reply:
I think it's as simple as `guice` publishes a bom and `asm` doesn't

Reply:
> I am not following why we add these specifically

Same.

Reply:
I think it's a matter of carefully removing dependencies, there's a PR to remove JNR: https://github.com/jenkinsci/jenkins/pull/5979

ASM comes via JNR right now, but removing ASM is another task which requires more changes.
So take steps at a time so that we can one day safely remove it, but currently plugins are getting asm libs from core so this means they don't get removed when JNR goes

Reply:
As Tim mentioned above, the reason is to keep the contents of the WAR identical with regard to the ASM JARs before and after the removal of the JNR JARs in #5979. Removing ASM from the WAR is trickier than removing JNR, so let's go one step at a time.

Reply:
> plugins like token macro depend on ASM. and I think scm-api does as well.

Core also depends on ASM via [Stapler](https://github.com/jenkinsci/stapler/blob/9a7b4b5916afe4ff0494698cf9ead253bdefd02c/core/src/main/java/org/kohsuke/stapler/ClassDescriptor.java#L28-L33) and [`access-modifier`](https://github.com/jenkinsci/lib-access-modifier/blob/b68c0122e7cfc4772fb2f8ab485b2157fdfa53f8/access-modifier-annotation/src/main/java/org/kohsuke/accmod/restrictions/ProtectedExternally.java#L64-L78), the latter of which [has declared its ASM dependency as `provided`](https://github.com/jenkinsci/lib-access-modifier/blob/b68c0122e7cfc4772fb2f8ab485b2157fdfa53f8/access-modifier-annotation/pom.xml#L19-L23) for five years.